opentelemetry-auto-instrumentation/src/opentelemetry/auto_instrumentation/instrumentor.py

```diff
@@ -38,14 +48,35 @@ def __new__(cls):
 
     @abstractmethod
     def _instrument(self, **kwargs):
-        """Instrument"""
+        """Instrument the library"""
 
     @abstractmethod
     def _uninstrument(self, **kwargs):
-        """Uninstrument"""
+        """Uninstrument the library"""
 
     def instrument(self, **kwargs):
-        """Instrument"""
+        """Instrument the library
+
+        This method will be called without any optional arguments by the
+        ``opentelemetry-auto-instrumentation`` command. The configuration of
+        the instrumentation when done in this way should be done by previously
+        setting the configuration (using environment variables or any other
+        mechanism) that will be used later by the code in the ``instrument``
+        implementation via the global ``Configuration`` object.
+
+        When this method is to be called directly in the user code, ``kwargs``
+        should be used to pass attributes that will override the configuration
+        values read by the ``Configuration`` object.
+
+        In this way, calling this method directly without passing any optional
+        values should do the very same thing that the
+        ``opentelemetry-auto-instrumentation`` command does. The idea behind
+        this approach is also to keep the ``Configuration`` object immutable.
+
+        This is necessary because this object is used by all the OpenTelemetry
```

Comment:
Might be good to call out that this paragraph refers to the Configuration object. I was confused as to whether this was referring to the Configuration object, the BaseInstrumentor class, or the auto-instrumentation command.

Reply:
Good observation, I have updated the documentation.